### PR TITLE
Update docs to explain purpose of SampleProjects

### PR DESCRIPTION
**Pull request by jgfoster**
This should avoid the impression that they are templates to be followed too closely. This is a partial fix for #149.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,10 +82,10 @@ vendor
 
 ### Installing the Dependencies
 
-Fulfilling the `arduino_ci` library dependency is as easy as running this command:
-
+Fulfilling the `arduino_ci` library dependency is as easy as running either of these two commands:
```
Comment (ianfixes) [inline README.md:85]: Conceptually, this is `bundle install`.  If you need an alternative that doesn't require admin access, `bundle install --vendor` is the second choice (aligning with #152) followed by `bundle install --path vendor/bundle/or/wherever`.

**Comment by jgfoster** [comment]
That's fine. I'm happy to provide feedback in whatever manner seems easier for you. I appreciate the quick turn-around on the ideas and questions. I'm learning a lot, making good progress, and will have more submissions for your consideration soon.

File: README.md (hunk 1 above)
Comment (jgfoster) [reply README.md:85]: On macOS 10.16, bundle version 1.17.2 gives `Unknown switches '--vendor'` in response to `bundle install --vendor`.  

As may be obvious, I'm not very familiar with Ruby, bundle, etc.

Comment (ianfixes) [reply README.md:85]: Sorry ... my head's in the wrong place. That should be `bundle install --deployment` 😅

File: README.md [hunk 2]
```diff
@@ -82,10 +82,10 @@ vendor
...
 ```
-$ bundle install
+$ bundle install   # adds packages to global library (may require admin rights)
+$ bundle install --path vendor/bundle   # adds packages to local library
```
Comment (ianfixes) [inline README.md:88]: Did you ever attempt `bundle install --deployment` ?  I'm curious if that worked

Comment (jgfoster) [reply README.md:88]: Yes, and it did not work. I reported the same error as a simple install (asking for a password and suggesting the local path as an alternative).

**Comment by jgfoster** [comment]
An unrelated matter (new issue?) but related to seeing things from the eyes of a new user:

I struggled with the compile vs. unit test options for a while. I spent a lot of time struggling with the compiler to get the unit tests to work and the idea that I'm running the unit tests without compiling (them?) seems very odd. Now that I'm understanding better I wonder if "Compile Examples" would be a better phrase (even though it is more verbose).

**Comment by ianfixes** [comment]
Lets's absolutely open that new issue.  `SampleProjects` becomes `CompileExamples`?  I really like that idea.

**Pull request merged**